pkg/app/rm.go

```diff
@@ -96,48 +96,71 @@ func (c *PolicyApp) Rm(existingRef string, force bool) error {
 	return nil
 }
 
-func (c *PolicyApp) removeFromIndex(ref *ocispec.Descriptor) error {
-
-	type index struct {
-		Version   int                  `json:"schemaVersion"`
-		Manifests []ocispec.Descriptor `json:"manifests"`
-	}
-
-	var localIndex index
-	indexPath := filepath.Join(c.Configuration.PoliciesRoot(), "index.json")
-	indexBytes, err := os.ReadFile(indexPath)
+func (c *PolicyApp) removeBasedOnManifest(ociClient *oci.Oci, ref *ocispec.Descriptor) error {
+	tarballDesc, configDesc, err := ociClient.GetTarballAndConfigLayerDescriptor(c.Context, ref)
 	if err != nil {
 		return err
 	}
 
-	err = json.Unmarshal(indexBytes, &localIndex)
+	// Hack to remove the existing digest until ocistore deleter is implemented
+	// https://github.com/oras-project/oras-go/issues/454
+	err = ociClient.GetStore().Delete(c.Context, *ref)
 	if err != nil {
 		return err
 	}
 
-	localIndex.Manifests = removeFromManifests(localIndex.Manifests, ref)
-
-	newIndexBytes, err := json.Marshal(localIndex)
+	tarballStillUsed, err := c.tarballReferencedByOtherManifests(ociClient, tarballDesc)
 	if err != nil {
 		return err
 	}
 
-	err = os.WriteFile(indexPath, newIndexBytes, 0664) //nolint:gosec,keep same permissions
+	if !tarballStillUsed {
+		err = ociClient.GetStore().Delete(c.Context, *tarballDesc)
+		if err != nil {
+			return err
+		}
+	}
+	err = ociClient.GetStore().Delete(c.Context, *configDesc)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func removeFromManifests(manifests []ocispec.Descriptor, ref *ocispec.Descriptor) []ocispec.Descriptor {
-	newarray := make([]ocispec.Descriptor, len(manifests)-1)
-	k := 0
-	for i := 0; i < len(manifests); i++ {
-		if manifests[i].Digest == ref.Digest && manifests[i].Annotations[ocispec.AnnotationRefName] == ref.Annotations[ocispec.AnnotationRefName] {
-			continue
+func (c *PolicyApp) tarballReferencedByOtherManifests(ociClient *oci.Oci, ref *ocispec.Descriptor) (bool, error) {
+	type index struct {
+		Version   int                  `json:"schemaVersion"`
+		Manifests []ocispec.Descriptor `json:"manifests"`
+	}
+
+	var localIndex index
+	indexPath := filepath.Join(c.Configuration.PoliciesRoot(), "index.json")
+	indexBytes, err := os.ReadFile(indexPath)
+	if err != nil {
+		return false, err
+	}
+
+	err = json.Unmarshal(indexBytes, &localIndex)
+	if err != nil {
+		return false, err
+	}
+	for i := range localIndex.Manifests {
```

Comment:
Could be replaced with
```go
	for _, descriptor := range localIndex.Manifests {
```

Reply:
Avoid  G601: Implicit memory aliasing in for loop. (gosec)